include/libtorrent/entry.hpp

```diff
@@ -95,8 +95,28 @@ namespace libtorrent
 			bool operator()(T1 const& rhs, T2 const& lhs) const
 			{ return rhs < lhs; }
 		};
-	};
+
+		template<class T> using map_string = std::map<std::string, T, aux::strview_less>;
+#else
+		template<class T>
+		class map_string : public std::map<std::string, T>
+		{
+		private:
+			using base = std::map<std::string, T>;
+		public:
+
+			typename base::iterator find(const string_view& key)
```

Comment:
builds without problem. 
the insert to emplace change was made for optimisation purpose.